File: snippets/fsharp/System/Double/NaN/double.nan4.fs
module double.nan4

// <Snippet4>
open System

printfn $"NaN = NaN: {Double.NaN = Double.NaN}"
printfn $"NaN <> NaN: {Double.NaN <> Double.NaN}"
printfn $"NaN.Equals(NaN): {Double.NaN.Equals Double.NaN}"
printfn $"not (NaN.Equals NaN): {not (Double.NaN.Equals Double.NaN)}"
printfn $"IsNaN: {Double.IsNaN Double.NaN}"

printfn $"\nNaN > NaN: {Double.NaN > Double.NaN}"
printfn $"NaN >= NaN: {Double.NaN >= Double.NaN}"
printfn $"NaN < NaN: {Double.NaN < Double.NaN}"
printfn $"NaN < 100.0: {Double.NaN < 100.0}"
printfn $"NaN <= 100.0: {Double.NaN <= 100.0}"
printfn $"NaN >= 100.0: {Double.NaN > 100.0}"
printfn $"NaN.CompareTo(NaN): {Double.NaN.CompareTo Double.NaN}"
printfn $"NaN.CompareTo(100.0): {Double.NaN.CompareTo 100.0}"
printfn $"(100.0).CompareTo(Double.NaN): {(100.0).CompareTo Double.NaN}"
// The example displays the following output:
//       NaN = NaN: False
//       NaN <> NaN: True
//       NaN.Equals(NaN): True
//       not (NaN.Equals NaN): False
//       IsNaN: True
//
//       NaN > NaN: False
//       NaN >= NaN: False
//       NaN < NaN: False
//       NaN < 100.0: False
//       NaN <= 100.0: False
//       NaN >= 100.0: False
//       NaN.CompareTo(NaN): 0
//       NaN.CompareTo(100.0): -1
//       (100.0).CompareTo(Double.NaN): 1
// </Snippet4>

File: snippets/csharp/System/Double/NaN/double.nan4.cs
// <Snippet4>
using System;

public class Example
{
    public static void Main()
    {
        Console.WriteLine($"NaN == NaN: {double.NaN == double.NaN}");
        Console.WriteLine($"NaN != NaN: {double.NaN != double.NaN}");
        Console.WriteLine($"NaN.Equals(NaN): {double.NaN.Equals(double.NaN)}");
        Console.WriteLine($"! NaN.Equals(NaN): {!double.NaN.Equals(double.NaN)}");
        Console.WriteLine($"IsNaN: {double.IsNaN(double.NaN)}");

        Console.WriteLine($"\nNaN > NaN: {double.NaN > double.NaN}");
        Console.WriteLine($"NaN >= NaN: {double.NaN >= double.NaN}");
        Console.WriteLine($"NaN < NaN: {double.NaN < double.NaN}");
        Console.WriteLine($"NaN < 100.0: {double.NaN < 100.0}");
        Console.WriteLine($"NaN <= 100.0: {double.NaN <= 100.0}");
        Console.WriteLine($"NaN >= 100.0: {double.NaN > 100.0}");
        Console.WriteLine($"NaN.CompareTo(NaN): {double.NaN.CompareTo(double.NaN)}");
        Console.WriteLine($"NaN.CompareTo(100.0): {double.NaN.CompareTo(100.0)}");
        Console.WriteLine($"(100.0).CompareTo(Double.NaN): {(100.0).CompareTo(double.NaN)}");
    }
}
// The example displays the following output:
//       NaN == NaN: False
//       NaN != NaN: True
//       NaN.Equals(NaN): True
//       ! NaN.Equals(NaN): False
//       IsNaN: True
//
//       NaN > NaN: False
//       NaN >= NaN: False
//       NaN < NaN: False
//       NaN < 100.0: False
//       NaN <= 100.0: False
//       NaN >= 100.0: False
//       NaN.CompareTo(NaN): 0
//       NaN.CompareTo(100.0): -1
//       (100.0).CompareTo(Double.NaN): 1
// </Snippet4>
